### PR TITLE
task/DES-2702: Tapis v3 Auth

### PR DESCRIPTION
**Pull request by rstijerina**
## Overview: ##

Adds `tapipy` package, Tapis v3 OAuth Token model, and v3 authorization flow

## PR Status: ##

* [X] Ready.

## Related Jira tickets: ##

* [DES-2702](https://tacc-main.atlassian.net/browse/DES-2702)

## Testing Steps: ##
1. Update `designsafe.env` with https://stache.utexas.edu/entry/ee36851448345135988a15c0684464ce to get new `TAPIS_` settings
2. `make build && make start`
3. `./manage.py migrate` in `des_django`
4. Navigate to https://designsafe.dev/login/ and confirm login works

## UI Images:
You should be redirected to the Tapis OAuth login screen:
Screenshot 2024-02-28 at 3 40 44 PM

Then after successful login, redirected back to the profile:
Screenshot 2024-02-28 at 3 42 19 PM

## Notes: ##
This includes breaking changes to the `main` branch. I propose we point this to a feature branch for v3 development.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -6,7 +6,6 @@ authors = ["DesignSafe-CI <designsafe-ci@tacc.utexas.edu>"]
 
 [tool.poetry.dependencies]
 python = "^3.11"
-agavepy = "1.0.0a12"
 pytas = {git = "https://bitbucket.org/taccaci/pytas.git", tag = "v1.7.0"}
 attrdict = { git = "https://github.com/DesignSafe-CI/AttrDict", rev = "83b779ee82d5b0e33be695d398162b8f2430ff33" }
```
Comment (nathanfranklin) [inline pyproject.toml:10]:
```suggestion
```
this was to workaround issues with agavepy so could be dropped with agavepy

Comment (rstijerina) [reply pyproject.toml]: Thanks, removed

File: designsafe/apps/auth/urls.py [hunk 2]
```diff
@@ -1,36 +1,42 @@
...
-                'visible': False
-            },
-            {
-                'label': _('Agave'),
-                'url': reverse('designsafe_auth:agave_session_error'),
-                'children': [],
-                'visible': False
-            }
-        ]
+# from django.urls import reverse
```
Comment (nathanfranklin) [inline designsafe/apps/auth/urls.py:18]: drop?  or todo comment?

Comment (rstijerina) [reply designsafe/apps/auth/urls.py]: good note, removed.

**Review by nathanfranklin — APPROVED** [review]
LGTM.  👍 

I did not run locally.

**Pull request merged**